cdk/lib/support-workers.ts

```diff
@@ -0,0 +1,236 @@
+}
+export class SupportWorkers extends GuStack {
+  constructor(scope: App, id: string, props: SupportWorkersProps) {
+    super(scope, id, props);
+
+    const app = "support-workers";
+
+    const lambdaDefaultConfig: Pick<
+      GuFunctionProps,
+      | "app"
+      | "memorySize"
+      | "fileName"
+      | "runtime"
+      | "timeout"
+      | "environment"
+      | "architecture"
+    > = {
+      app,
+      memorySize: 1536,
+      fileName: `support-workers.jar`,
+      runtime: Runtime.JAVA_21,
+      timeout: Duration.seconds(300),
+      architecture: Architecture.ARM_64,
+      environment: {
+        APP: app,
+        STACK: this.stack,
+        STAGE: this.stage,
+        SENTRY_DSN:
+          "https://55945d73ad654abd856d1523de4f9d56:cf9b33aaff3c483899dfa986abce55df@sentry.io/1212214",
+        SENTRY_ENVIRONMENT: this.stage,
+        GU_SUPPORT_WORKERS_STAGE: this.stage,
+        EMAIL_QUEUE_NAME: Fn.importValue(`comms-${this.stage}-EmailQueueName`),
+      },
+    };
+
+    const s3Policy = new PolicyStatement({
+      actions: ["s3:GetObject"],
+      resources: props.s3Files,
+    });
+    const emailSqsPolicy = new PolicyStatement({
+      actions: ["sqs:GetQueueUrl", "sqs:SendMessage"],
+      resources: [Fn.importValue(`comms-${this.stage}-EmailQueueArn`)],
+    });
+    const eventBusPolicy = new PolicyStatement({
+      actions: ["events:PutEvents"],
+      resources: props.eventBusArns,
+    });
+    const cloudWatchLoggingPolicy = new PolicyStatement({
+      actions: [
+        "logs:Create*",
+        "logs:PutLogEvents",
+        "logs:DescribeLogStreams",
+        "cloudwatch:putMetricData",
+      ],
+      resources: ["*"],
+    });
+    const promotionsDynamoTablePolicy = new PolicyStatement({
+      actions: [
+        "dynamodb:GetItem",
+        "dynamodb:Scan",
+        "dynamodb:Query",
+        "dynamodb:DescribeTable",
+      ],
+      resources: props.promotionsDynamoTables,
+    });
+    const supporterProductDataTablePolicy = new PolicyStatement({
+      actions: ["dynamodb:PutItem", "dynamodb:UpdateItem"],
+      resources: props.supporterProductDataTables.map((table) =>
+        Fn.importValue(table)
+      ),
+    });
+
+    const createLambda = (
+      lambdaName: string,
+      additionalPolicies: PolicyStatement[] = []
+    ) => {
+      const lambdaId = `${lambdaName}Lambda`;
+      const lambda = new GuLambdaFunction(this, lambdaId, {
+        ...lambdaDefaultConfig,
+        handler: `com.gu.support.workers.lambdas.${lambdaName}::handleRequest`,
+        functionName: `${this.stack}-${lambdaName}Lambda-${this.stage}`,
+        environment: {
+          ...lambdaDefaultConfig.environment,
+        },
+        initialPolicy: [
+          s3Policy,
+          cloudWatchLoggingPolicy,
+          ...additionalPolicies,
+        ],
+      });
+      this.overrideLogicalId(lambda, {
+        logicalId: lambdaId,
+        reason: "Moving existing lambda to CDK",
+      });
```

Comment:
that's a clever trick, it's always tricky to get around this kind of auto generated IDs thing normally.  I assume this is a permanent override though?
```suggestion
      this.overrideLogicalId(lambda, {
        logicalId: lambdaId,
        reason: "Retain pre-CDK logical ID",
      });
```